### DISCUSSION TITLE
png logo into SVG QR Code

### DISCUSSION THREAD

**Discussion by AmjadGhzlangit** [chillerlan/php-qrcode#268] [Q&A]
Hi, how to put a  png logo into SVG QR Code

**Comment by codemasher** [comment]
You can embed raster images in SVG by using the `<image>` element: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image

Basically, you'd just have to change the code in the example so that it loads a snippet with the image tag instead of another SVG file

https://github.com/chillerlan/php-qrcode/blob/0ca3d2377f8d01b788c738d46f9bf72d9b17083d/examples/svgWithLogo.php#L52-L94

**Comment by oskay12** [comment]
i asked to much but thats also important for me.
when i try to put an svg logo if its not modified logo is not appearing at the attachment alamy is working but other one not
![alamy](https://github.com/user-attachments/assets/d16296f2-e67b-4fef-b651-05c5573765e0)
![github-one-svgrepo-com](https://github.com/user-attachments/assets/66150f22-dcc4-47ab-afc0-385472d39eca)

**Reply by codemasher** [reply]
I'm not sure if I understand correctly: you want to embed an SVG logo but it is not positioned correctly? (btw. this isn't the proper thread then, please open separate discussions in the future)

Generally you need to wrap the SVG logo in a `<g>` element and move it to the desired position proportionally using the `transform` attribute.

The example should do this automatically here:
https://github.com/chillerlan/php-qrcode/blob/81291a4b6b336a08468de827d24ec5f13be7cb3f/examples/svgWithLogoAndCustomShapes.php#L119

See:
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/g
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform

(if you want to embed a raster image such as PNG, this is answered above)
